esp/esdllib/esdl_script.cpp

```diff
@@ -771,111 +829,129 @@ class CEsdlTransformOperationHttpPostXml : public CEsdlTransformOperationBase
-            targetContext->addXmlContent(xml.str());
+            me->serialize(exceptionXml);
             me->Release();
         }
         catch(IException *E)
         {
-            StringBuffer xml;
             Owned<IMultiException> me = makeMultiException("ESDLScript");
             me->append(*LINK(E));
-            me->serialize(xml);
-            CXpathContextLocation content_location(targetContext);
-            targetContext->ensureSetValue("@status", "error", true);
-            targetContext->addElementToLocation("content");
-            targetContext->addXmlContent(xml.str());
+            me->serialize(exceptionXml);
             E->Release();
         }
-    }
 
-    virtual bool process(IEsdlScriptContext * scriptContext, IXpathContext * targetContext, IXpathContext * sourceContext) override
-    {
-        VStringBuffer xpath("/esdl_script_context/%s/%s", m_section.str(), m_name.str());
+        if (preparedState.testDelay)
+            MilliSleep(preparedState.testDelay);
+
+        //No need to synchronize until we hit this point and start updating the scriptContext / xml document
+        OptionalCriticalBlock block(crit);
+
         CXpathContextLocation location(targetContext);
+        VStringBuffer xpath("/esdl_script_context/%s/%s", m_section.str(), m_name.str());
         targetContext->ensureLocation(xpath, true);
-        StringBuffer url;
-        if (m_url)
-            sourceContext->evaluateAsString(m_url, url);
 
-        Owned<IProperties> headers = createProperties();
-        buildRequest(scriptContext, targetContext, sourceContext, url, headers);
-        postRequest(scriptContext, targetContext, sourceContext, url, headers);
+        if (exceptionXml.length())
+        {
+            CXpathContextLocation content_location(targetContext);
+            targetContext->addElementToLocation("content");
+            targetContext->ensureSetValue("@status", "error", true);
+            targetContext->addXmlContent(exceptionXml.str());
+        }
+        else
+        {
+            CXpathContextLocation response_location(targetContext);
+            targetContext->addElementToLocation("response");
+            targetContext->ensureSetValue("@status", status.str(), true);
+            targetContext->ensureSetValue("@error-code", err.str(), true);
+            targetContext->ensureSetValue("@content-type", contentType.str(), true);
+            if (strnicmp("text/xml", contentType.str(), 8)==0 || strnicmp("application/xml", contentType.str(), 15) ==0)
+            {
+                CXpathContextLocation content_location(targetContext);
+                targetContext->addElementToLocation("content");
+                targetContext->addXmlContent(response.str());
+            }
+            else
+            {
+                targetContext->ensureSetValue("text", response.str(), true);
+            }
+        }
 
         sourceContext->addXpathVariable(m_name, xpath);
```

Comment:
Unrelated, but it seems strange that the source context is updated.

Reply:
Yes currently you could say sourceContext is badly named.  It is currently "sourceAndVariableContext"  or maybe "sourceAndCodeContext".

Variables, scopes, etc. are managed through that object, as well as the proper read only source context stuff.

I may refactor at some point to separate them, but not in this PR.